lib/elixir/lib/option_parser.ex

```diff
@@ -394,16 +394,27 @@ defmodule OptionParser do
       ["--bool", "--no-bool"]
 
   """
-  @spec to_argv(Enumerable.t) :: argv
-  def to_argv(enum) do
+  @spec to_argv(Enumerable.t, options) :: argv
+  def to_argv(enum, opts \\ []) do
+    switches = Keyword.get(opts, :switches, [])
     Enum.flat_map(enum, fn
       {_key, nil}  -> []
       {key, true}  -> [to_switch(key)]
       {key, false} -> [to_switch(key, "--no-")]
-      {key, value} -> [to_switch(key), to_string(value)]
+      {key, value} -> to_argv(key, value, switches)
     end)
   end
 
+  defp to_argv(key, value, switches) do
+    if switches[key] == :count do
+      List.duplicate(to_switch(key), value)
+    else
+      # If no options were provided consider the argv a simple pair
+      [to_switch(key), to_string(value)]
+    end
+  end
+
+
```

Comment:
Sorry one last comment, there is an extra new line here. Other than that, we are good to go! Thank you!